ledger/src/blockstore.rs

```diff
@@ -1838,6 +1852,72 @@ impl Blockstore {
             .map(|x| x.0)
     }
 
+    fn verify_shred(&self, leader: &Pubkey, slot: Slot, x: u64) {
+        let payload = self.get_data_shred(slot, x).unwrap().unwrap();
+        let shred = Shred::new_from_serialized_shred(payload).unwrap();
+        assert!(
+            shred.verify(leader),
+            "shred failed verification {} {} {}",
+            slot,
+            x,
+            self.shreds_dir,
+        );
+    }
+
+    fn verify_expected_shreds(
+        &self,
+        ls: Option<&Arc<LeaderScheduleCache>>,
+        slot: Slot,
+    ) -> ShredIndex {
```

Comment:
this should check all the shreds in rocks against the slot directories @carllin

Reply:
do we also need to validate coding shreds? Or is it ok if coding recovery fails? Worst case a coding shred will just cause recovery to repeatedly fail here: https://github.com/solana-labs/solana/blob/master/ledger/src/blockstore.rs#L750-L763, doesn't seem too bad

Reply:
doesn't seem like we should panic on those?  i think v1 of this would be using erasure to recover any failed data shreds